### PR TITLE
feat: APP-2330 CSV export on transfers page implemented

### PR DESCRIPTION
**Pull request by babarella**
## Description

Export transfers as .csv file

Task: [APP-2330](https://aragonassociation.atlassian.net/browse/APP-2330)

## Type of change
- [x]  New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2330]: https://aragonassociation.atlassian.net/browse/APP-2330?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/locales/en/common.json [hunk 1]
```diff
@@ -773,7 +773,7 @@
       "minimumApprovalSubtitle": "Minimum approval is the amount of addresses in the authorized list that must approve a proposal for it to pass. ",
       "minApprovalAddressCount": "of {{count}} addresses",
       "alerts": {
-        "minimumApprovalAlert": "of {{amount}} {{tokenSymbol}}",
```
Comment (sepehr2github) [inline src/locales/en/common.json:776]: Why did you remove tokenSymbol Since we're still passing it

File: src/containers/exportCsvModal/ExportCsvModal.tsx [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+    setCsvBlob(undefined);
+    setIsCsvGenerationError(false);
+    setIsCsvGenerationSuccess(false);
+    setIsFlowFinished(false);
+    close('exportCsv');
+  }, [close]);
+
+  const handleFlowFinish = useCallback(() => {
+    handleClose();
+    navigate(
```
Comment (sepehr2github) [inline src/containers/exportCsvModal/ExportCsvModal.tsx]: I think Users should remain on the current page and not forcing them to visit finance, I was exporting in transfer page then I was sent to finance with no reason

File: src/containers/exportCsvModal/ExportCsvModal.tsx [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  const downloadCsvFile = useCallback(async () => {
+    if (!csvBlob) return;
+
+    try {
+      const fileName = `export-txns-${Math.floor(
+        Date.now() / 1000
+      )}-${getCanonicalDate().split('-').join('')}.csv`;
+
+      await saveAs(csvBlob, fileName);
+    } catch (e) {
```
Comment (sepehr2github) [inline src/containers/exportCsvModal/ExportCsvModal.tsx:169]: It seems that you can remove wait from here, thus the function doesn't need to be async

File: src/pages/transfers.tsx [hunk 4]
```diff
@@ -108,6 +137,22 @@ const Transfers: React.FC = () => {
                   label={t('labels.externalContract')}
                 /> */}
               </ButtonGroup>
+
+              {dropdownActions.length && (
```
Comment (sepehr2github) [inline src/pages/transfers.tsx]: Since the array is somehow hardcoded, I don't understand the intention behind this condition

**Review by sepehr2github — COMMENTED** [review]
Hey, Hey. Thanks Nikita. I know How frustrating working with CSV libraries could be, Left a few comments

File: src/containers/exportCsvModal/ExportCsvModal.tsx [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+
+        return transferATimestamp > transferBTimestamp
+          ? 1
+          : transferATimestamp < transferBTimestamp
+          ? -1
+          : 0;
+      });
+  }, [endDate, startDate, transfers]);
+
+  const csvData = useMemo(() => {
```
Comment (sepehr2github) [inline src/containers/exportCsvModal/ExportCsvModal.tsx:103]: Probably worth a sync with Selim but I'd prefer to have latest transfers on top of the list which means the current order should be reversed

File: src/locales/en/common.json (hunk 1 above)
Comment (babarella) [reply src/locales/en/common.json:776]: That's not me, that's the updated `common.json` file. I guess Selim updated it at some point of time.

File: src/containers/exportCsvModal/ExportCsvModal.tsx (hunk 3 above)
Comment (babarella) [reply src/containers/exportCsvModal/ExportCsvModal.tsx:169]: If I remove `await`, then `try-catch` will not work and catch potential errors.

**Review by sepehr2github — APPROVED** [review]
Thanks for fixing
